response.go

```diff
@@ -96,8 +94,23 @@ func (r *Response) Write(b []byte) (int, error) {
 			rc.Close()
 		}
 		r.Body = buf
-		return buf.Write(b)
+		n, err = buf.Write(b)
+		if err != nil {
+			return 0, err
```

Comment:
I don't think this is entirely correct, you can get back an error if `n < len(b)` but that doesn't mean that n is zero. We should return `n, err`